internal/pkg/csumio/csumwriter_test.go

```diff
@@ -0,0 +1,99 @@
+package csumio_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"testing"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/cybozu-go/fin/internal/pkg/csumio"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChecksumWriter(t *testing.T) {
+	// Description:
+	// Check that ChecksumWriter correctly writes data and generates checksums for various data lengths.
+	// Tests multiple scenarios: chunk-aligned data and non-aligned data.
+	//
+	// Arrange:
+	// - Prepare test data of various lengths relative to the chunk size
+	// - Create a ChecksumWriter with buffers for data and checksums
+	//
+	// Act:
+	// Write all data using ChecksumWriter, then close
+	//
+	// Assert:
+	// - Data should be written correctly to data buffer
+	// - Checksums should be generated correctly for all chunks
+
+	chunkSize := 4096 // 4KiB
```

Comment:
Please use minimumChunkSize here. In addition, please replace literal numbers in this test to expressions using chunkSize. The changes would be like this:

- 4096 -> chunkSize
- 4097 -> chunkSize+1
- 8192 -> chunkSize*2